CMakeLists.txt

```diff
@@ -13,6 +13,15 @@ find_package(everest-cmake 0.1 REQUIRED
 option(${PROJECT_NAME}_BUILD_TESTING "Build unit tests, used if included as dependency" OFF)
 option(BUILD_TESTING "Build unit tests, used if standalone project" OFF)
 option(EVSE_SECURITY_INSTALL "Install the library (shared data might be installed anyway)" ${EVC_MAIN_PROJECT})
+option(USING_TPM2 "Include code for using OpenSSL 3 and the tpm2 provider" OFF)
+
+if(USING_TPM2)
+    # OpenSSL property string when using the default provider
+    set(PROPQUERY_DEFAULT "provider!=tpm2")
```

Comment:
Question here, is it certain that based on the platform we might need to modify this property queries? If yes, based on what criteria? Isn't it better to have these hard-coded in the .cpp files, or there's a strong possibility that this might be required to be changed based on platform?

Reply:
There are other hardware (and firmware) security devices and approaches. Allowing the property strings to be configured could allow them to be supported. In my specific case I'm using the tmp2 provider as well as tpm2tss.
There is a pkcs11 provider that would support Hardware Security Modules (similar to a TPM).
A board might have a crypto accelerator chip which might need a slight change to the property string …